### PR TITLE
Add GitHub Action to create an issue containing a release checklist

### LINKED ISSUE #1596: Create GitHub Action that automatically creates an issue for releases with a release checklist
The text for this could be adapted from #1595.

The content from most of the release guide might be best ported into this file.

### PR DESCRIPTION
**Pull request by namurphy**
## Description

This PR adds a GitHub Action (mostly created by ChatGPT!) that lets us create an issue for a release that contains a release checklist.

## Motivation and context

We currently have a release guide, but it contains a lot of steps so it can be easy to forget something.  By having a checklist, we can help the process go more smoothly.

My hope is that this checklist can entirely replace the release guide (which would essentially be reduced to instructions on how to create a release checklist issue).

## Related issues

Closes #1596, with the caveat that it won't be automatic since we'll still have to trigger the action manually.

### CODE REVIEW

**Comment by namurphy** [comment]
Oops, I included some changes to GitHub Actions that I did not mean to include.  I'll need to spin those changes off into a separate pull request to keep the pull request size more manageable.

File: .github/workflows/weekly-tests.yml [hunk 1]
```diff
@@ -72,11 +72,6 @@ jobs:
           toxenv: build_docs-sphinxdev
           toxposargs: -q
```
Comment (namurphy) [inline .github/workflows/weekly-tests.yml:73]:
```suggestion
          toxposargs: -q

        - name: Documentation build with hyperlink check
          os: ubuntu-latest
          python: '3.11'
          toxenv: build_docs_linkcheck
```

File: .github/workflows/create-release-issue.yml
Comment (namurphy) [inline .github/workflows/create-release-issue.yml:1]: Co-authored-by: ChatGPT 🐱

Comment (namurphy) [reply .github/workflows/create-release-issue.yml:1]: ...though I guess with the emoji it'd be CatGPT?

Comment (ejohnson-96) [reply .github/workflows/create-release-issue.yml:1]: -____________________________-

**Review by ejohnson-96 — APPROVED** [review]
👍

**Pull request merged**